### PR TITLE
Feature/drop zone

### PR DESCRIPTION
**Pull request by kristofferj**

### CODE REVIEW

File: packages/@sanity/components/src/fileinput/DropZone.js [hunk 1]
```diff
@@ -1,43 +1,143 @@
...
+    accept: PropTypes.string
   }
 
   static defaultProps = {
     multiple: true
   }
 
+  state = {
+    hasUnAcceptedFiles: false,
+    toManyFiles: false
```
Comment (bjoerge) [inline packages/@sanity/components/src/fileinput/DropZone.js]: Typo: `tooMany`

File: packages/@sanity/components/src/fileinput/DropZone.js [hunk 2]
```diff
@@ -1,43 +1,143 @@
...
+    ghost: PropTypes.bool,
+    accept: PropTypes.string
   }
 
   static defaultProps = {
     multiple: true
   }
 
+  state = {
+    hasUnAcceptedFiles: false,
```
Comment (bjoerge) [inline packages/@sanity/components/src/fileinput/DropZone.js]: Can this be changed to `acceptsAll: true`?

File: packages/@sanity/components/src/fileinput/DropZone.js [hunk 3]
```diff
@@ -1,43 +1,143 @@
...
+      if (items.length > 1 && !this.props.multiple) {
+        this.setState({
+          toManyFiles: true
+        })
+        return
+      }
+
+      for (let i = 0; i < items.length; i++) {
+        this.setState({
+          hasUnAcceptedFiles: !includes(mimeTypes, items[i].type)
```
Comment (bjoerge) [inline packages/@sanity/components/src/fileinput/DropZone.js]: No need for lodash here, you can do: `mimeTypes.includes(items[i].type)`

Comment (bjoerge) [inline packages/@sanity/components/src/fileinput/DropZone.js]: Also, if the `accept` prop is supposed to be a stand in for the `accept` attribute on native file inputs, this check is too simple. Could we use this instead? https://github.com/okonet/attr-accept

File: packages/@sanity/components/src/fileinput/DropZone.js (hunk 2 above)
Comment (kristofferj) [inline packages/@sanity/components/src/fileinput/DropZone.js]: Later in the code I check if there is files that is not accepted. Setting error state based on acceptsAll=false is confusing.

File: packages/@sanity/components/src/fileinput/DropZone.js (hunk 1 above)
Comment (kristofferj) [inline packages/@sanity/components/src/fileinput/DropZone.js]: 👍

File: packages/@sanity/components/src/fileinput/DropZone.js (hunk 3 above)
Comment (kristofferj) [inline packages/@sanity/components/src/fileinput/DropZone.js]: Yep. I was hoping somebody could do a proper check here.

Comment (bjoerge) [inline packages/@sanity/components/src/fileinput/DropZone.js]: By looking closer the for loop could also be rewritten to:
```js
this.setState({acceptsAll: items.every(item => mimeTypes.includes(item))})
```

Would you like to give attr-accept a try?

**Comment by bjoerge** [comment]
Fantastic 💖

**Pull request merged**